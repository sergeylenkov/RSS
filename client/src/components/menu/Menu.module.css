.container {
    position: fixed;
	margin: 0;
	display: flex;
	flex-wrap: wrap;
	flex-direction: column;	
	top: 40px;
	left: 30px;
}

.feeds {
    margin: 20px 0 0 0;
    display: flex;
	flex-wrap: wrap;
	flex-direction: column;
}