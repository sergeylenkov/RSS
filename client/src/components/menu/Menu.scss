.menu {
	&__container {
		display: flex;
		height: 100%;
		margin: 0 auto 0 0;
	}
}


@media (max-width: 768px) {
	.menu {
		&__container {
			overflow: auto;
		}
	}
}
