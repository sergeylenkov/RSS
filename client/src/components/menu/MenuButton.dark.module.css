.container {
    composes: container from './MenuButton.module.css';
}

.label {
	composes: label from './MenuButton.module.css';
	color: #bfbfbf;
}

.container.selected .label {
	font-weight: bold;
}

.counter {
	composes: counter from './MenuButton.module.css';
	color: #ffffff;
    background-color: #4da0da;
}

.disabled {
	composes: disabled from './MenuButton.module.css';
}