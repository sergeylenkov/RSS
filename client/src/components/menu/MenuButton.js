import React from 'react';
import { Icon, Icons } from '../Icon.js';

import styles from './MenuButton.module.css';

export class MenuButton extends React.Component {
    render() {
        const feed = this.props.feed;
        const isEdited = this.props.isEdited;

        let className = styles.container;

        if (isEdited) {
            className += ` ${styles.edited}`;

            return (
                <button className={className}>
                    <div className={styles.icon} style={{ backgroundImage: `url(${feed.icon})` }}></div>
                    <div className={styles.label}>{feed.title}</div>
                    <button className={styles.deleteButton} onClick={() => this.onDelete()}><Icon svg={Icons.trash}/></button>                    
                </button>
            );
        } else {
            if (feed.count === 0) {
                className += ` ${styles.empty}`;
            }
    
            if (this.props.isSelected) {
                className += ` ${styles.selected}`;
            }

            return (
                <button className={className} onClick={() => this.onClick()}>
                    <div className={styles.icon} style={{ backgroundImage: `url(${feed.icon})` }}></div>
                    <div className={styles.label}>{feed.title}</div>
                    <div className={styles.counter}>{feed.count}</div>
                </button>
            );        
        }
    }

    onClick() {
        this.props.onClick(this.props.feed.id);
    }

    onDelete() {
        this.props.onDelete(this.props.feed.id);
    }
}