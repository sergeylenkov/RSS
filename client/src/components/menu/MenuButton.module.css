.container {
	height: 100%;
	margin: 0 0 0 20px;
	display: flex;
	flex-direction: row;
	cursor: pointer;
}

.label {
	margin: auto 0;
	font-size: 14px;
}

.container.selected .label {
	font-weight: bold;
}

.counter {
	margin: auto 0 auto 10px;
	padding: 2px 5px;
	font-size: 11px;
	border-radius: 2px;
	color: #ffffff;
    background-color: #4da0da;
}

.disabled {
	pointer-events: none;
}