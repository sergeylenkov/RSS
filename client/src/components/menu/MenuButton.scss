.menu-button {
  display: flex;
  flex-direction: row;
  flex-shrink: 0;
  height: 100%;
  margin: 0 20px 0 0;
  cursor: pointer;

  &__label {
    margin: auto 0;
    font-size: 14px;
    color: #000000;

    &-selected {
      font-weight: bold;
    }
  }

  &__counter {
    margin: auto 0 auto 10px;
    padding: 2px 5px;
    font-size: 11px;
    border-radius: 2px;
    color: #ffffff;
    background-color: #4da0da;
  }
}

/*
* theme
*/

.dark .menu-button {
  &__label {
    color: #bfbfbf;

    &_selected {
      color: #f3f3f3;
    }
  }

  &__counter {
    color: #ffffff;
    background-color: #4da0da;
  }
}
