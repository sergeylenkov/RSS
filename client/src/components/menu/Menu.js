import React from 'react';
import { ReloadButton } from './ReloadButton.js';
import { MenuButton } from './MenuButton.js';
import { connect } from 'react-redux';

import styles from './Menu.module.css';

class ConnectedMenu extends React.Component {
    constructor(props) {
        super(props);

        this.onUpdate = this.onUpdate.bind(this);
    }

    render() {
        return (
            <div className={styles.container}>
                <ReloadButton isActive={this.props.isUpdating} isSelected={this.props.type === 0} count={this.props.count} onClick={this.onUpdate}/>
                <MenuButton title={'Все'} isSelected={this.props.type === 1} onClick={this.props.onShowAll} />
                <MenuButton title={'Прочитанное'} isSelected={this.props.type === 2} onClick={this.props.onShowRead} />
                <MenuButton title={'Избранное'} isSelected={this.props.type === 3} onClick={this.props.onShowBookmark}/>
            </div>
        );
    }

    onUpdate() {        
        this.props.updateFeeds();
        this.props.onUpdate();
    }
}

/* Redux */

const mapStateToProps = state => {
    return {
        isUpdating: state.isUpdating
    };
};

export const Menu = connect(mapStateToProps)(ConnectedMenu);