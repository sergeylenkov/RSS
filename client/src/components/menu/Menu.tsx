import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useSelector } from 'react-redux'
import MenuButton from './MenuButton';
import ReloadButton from './ReloadButton';
import { Bem } from '../../utils/bem';
import { State } from '../../store/reducers';

import './Menu.scss';

const block = new Bem('menu');

interface MenuProps {
  onUpdate: () => void;
}

function Menu({ onUpdate }: MenuProps): JSX.Element {
  const history = useHistory();
  const { pathname } = useLocation();
  const isUpdating = useSelector<State, boolean>(state => state.isUpdating);
  const isUpdateError = useSelector<State, boolean>(state => state.isUpdateError);
  const unviewedCount = useSelector<State, number>(state => state.unviewedCount);
  const entriesCount = useSelector<State, number>(state => state.entriesCount);

  const onClick = (path: string) => {
    history.push(path);
  }

  return (
    <div className={block.build()}>
      <ReloadButton
        isActive={isUpdating}
        isSelected={pathname === '/'}
        isError={isUpdateError}
        count={unviewedCount}
        onUpdate={onUpdate}
        onClick={() => onClick('/')}
      />
      <MenuButton title={'Все'} isSelected={pathname === '/all'} count={entriesCount} onClick={() => onClick('/all')} />
      <MenuButton title={'Прочитанное'} isSelected={pathname === '/read'} count={entriesCount} onClick={() => onClick('/read')} />
      <MenuButton title={'Избранное'} isSelected={pathname === '/favorites'} count={entriesCount} onClick={() => onClick('/favorites')} />
    </div>
  );
}

export default Menu;
