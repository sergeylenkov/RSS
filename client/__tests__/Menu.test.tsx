import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../src/store';
import { Menu } from '../src/components/menu/Menu';

test('Menu', () => {
  const render = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    </Provider>
  );
  expect(render).toMatchSnapshot();
});
