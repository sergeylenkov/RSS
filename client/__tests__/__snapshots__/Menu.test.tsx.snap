// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu correct rendering 1`] = `
<div
  className="menu"
>
  <div
    className="reload-button"
  >
    <button
      className="reload-button__icon"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
      >
        <g>
          <polyline
            points="23 4 23 10 17 10"
          />
          <polyline
            points="1 20 1 14 7 14"
          />
          <path
            d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"
          />
        </g>
      </svg>
    </button>
    <button
      className="menu-button"
      onClick={[Function]}
    >
      <div
        className="menu-button__label menu-button__label_selected"
      >
        Свежее
      </div>
      <div
        className="menu-button__counter"
      >
        0
      </div>
    </button>
  </div>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <div
      className="menu-button__label"
    >
      Все
    </div>
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <div
      className="menu-button__label"
    >
      Прочитанное
    </div>
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <div
      className="menu-button__label"
    >
      Избранное
    </div>
  </button>
</div>
`;
